### PR TITLE
fix: make LocalStorage adapter available in the browser

### PR DESCRIPTION
**Pull request by Arondight**

### CODE REVIEW

**Comment by Arondight** [comment]
https://github.com/typicode/lowdb/pull/528 cause this issue, it breaks [Browser example](https://github.com/typicode/lowdb/tree/main/examples#browser).

**Comment by Arondight** [comment]
@typicode hi can you review this pull?

**Review by UlisesGascon — APPROVED** [review]

File: src/adapters/LocalStorage.ts [hunk 1]
```diff
@@ -2,13 +2,30 @@ import { SyncAdapter } from '../LowSync.js'
...
+    }
+
+    if (typeof window !== 'undefined' && 'localStorage' in window) {
+      this.#localStorage = window.localStorage
+    }
   }
 
   read(): T | null {
-    const value = global.localStorage.getItem(this.#key)
+    const value = this.#localStorage.getItem(this.#key)
```
Comment (FossPrime) [inline src/adapters/LocalStorage.ts:28]: it's better to define localStorage as a global in your ambient types. https://bobbyhadz.com/blog/typescript-declare-global-variable

probably also best to use globalThis, but I can't think of a context where that matters and you can use localStorage simultaneously.

**Comment by typicode** [comment]
Hey,

Sorry for the late reply.

As part of v4/v5 update, I went ahead and simply removed `global`. I tested it and it works in Vite and CodeSandbox (with .js).
Also TypeScript doesn't seem to complain when `lib` has `DOM` in `tsconfig.json`.

Is there still a failing use case that I may be missing? Or v5 fixes your initial issue?

**Comment by Arondight** [comment]
New version looks good but import path of adapters breaks old code

**Pull request closed without merging**